### ISSUE TITLE
Improve user experience with custom icon packs

### ISSUE THREAD

**Issue by FlorianWoelki** [FlorianWoelki/obsidian-icon-folder#224]
Due to the new syncing mechanisms, all icon packs will have to be zipped into a file so that the syncing works smoothly. This behaviour works with the predefined icon packs but not with the custom created icon packs.

The improvement will include a better user experience to support icon additions to a custom icon pack through a zip file.

This issue was created based on #223

**Comment by binaryDiv** [comment]
Hey @FlorianWoelki, just a quick question about this for my understanding: Does this issue mean that currently the "sync via zip file" feature does not work at all with custom icon packs? Or can I already add a zip file to a custom icon pack?

I'm asking because I'm experiencing the syncing issues described in #52 and just updated to 2.2.1 and am confused about how to use the new sync feature. ^^

Also, the only reason I'm using a custom icon pack actually is that the version of Lucide Icons in the predefined icon packs is quite outdated and a lot of new icons are missing in it. Are there plans to update the icon packs soon? I'll also gladly do a pull request for that if wanted. :)

**Comment by FlorianWoelki** [comment]
> Hey @FlorianWoelki, just a quick question about this for my understanding: Does this issue mean that currently the "sync via zip file" feature does not work at all with custom icon packs? Or can I already add a zip file to a custom icon pack?
> 
> I'm asking because I'm experiencing the syncing issues described in #52 and just updated to 2.2.1 and am confused about how to use the new sync feature. ^^
> 
> Also, the only reason I'm using a custom icon pack actually is that the version of Lucide Icons in the predefined icon packs is quite outdated and a lot of new icons are missing in it. Are there plans to update the icon packs soon? I'll also gladly do a pull request for that if wanted. :)

Yes you can already add a custom zip of your own custom icon pack and then use it. :)

I agree with the outdated icon packs, you can feel free to create a PR for that 🚀

**Comment by FlorianWoelki** [comment]
> just to be clear, this is not yet implemented per the `FlorianWoelki moved this to Todo in Iconize Board 3 days ago` bullet above? If so, are we currently unable to use custom icon packs with Obsidian sync? I just tried a zipped package and it didn't show in the sync logs. Workaround right now is using Syncthing to sync the icon pack to my phone

It should be possible to sync custom zip files in Obsidian sync. Maybe the zip file was empty?

**Comment by FlorianWoelki** [comment]
> Thanks for the reply @FlorianWoelki . No, I personally zipped the files myself, and confirmed it was not empty

Do you have something so that I can reproduce the issue? Because atm, I cannot reproduce it locally.

**Comment by FlorianWoelki** [comment]
> [zipped_custom_icons_14.zip](https://github.com/FlorianWoelki/obsidian-iconize/files/12652604/zipped_custom_icons_14.zip) Attached is my zipped 14 icon custom pack. I drop this into /.obsidian/icons, and confirm it shows in the plugin settings as a custom pack. But it never shows syncing to my Vault, and subsequently never arrives on my phone. I've deleted and re-added multiple times, even deleting the vault from my phone.

Thank you! That's unfortunate. Do you use Obsidian Sync? Because obviously, normal syncing would mean it thinks when something gets added/updated :D You could also try using [this approach](https://github.com/FlorianWoelki/obsidian-iconize?tab=readme-ov-file#sync-issue)

**Comment by FlorianWoelki** [comment]
> Yeah, I used to use Syncthing but now I have started paying for Obsidian Sync. And yes, I have already put my icons in `.obsidian/icons` but that does not solve the issue :/ Obsidian support doesn't have any answer to why this is happening, they seem to think it's something to do with the plugin. They said they've reached out to you but not response. Btw, I am just relaying the message here, not blaming you. I am just trying to get to the root of the problem :)

Well, they've never reached out to me, honestly. Then I need to reach out to them...

**Comment by lishid** [comment]
For Obsidian Sync, any non-recognized folders in `.obsidian` are ignored. The setting for syncing 3rd party plugins will sync files in the folders `.obsidian/plugins/plugin-name`, so this is why

1. You should **not** be storing thousands of icons in that folder, to avoid clogging up the syncing process. I've suggested [changing the default unpacked folder to `.obsidian/icons/`, but it seems like that hasn't happened yet](https://github.com/FlorianWoelki/obsidian-iconize/issues/52#issuecomment-1079740583).
2. If you do have a packed zip file, then you'll want to put it in the plugin folder so it gets synced.

**Comment by FlorianWoelki** [comment]
> For Obsidian Sync, any non-recognized folders in `.obsidian` are ignored. The setting for syncing 3rd party plugins will sync files in the folders `.obsidian/plugins/plugin-name`, so this is why
> 
> 1. You should **not** be storing thousands of icons in that folder, to avoid clogging up the syncing process. I've suggested [changing the default unpacked folder to `.obsidian/icons/`, but it seems like that hasn't happened yet](https://github.com/FlorianWoelki/obsidian-iconize/issues/52#issuecomment-1079740583).
> 2. If you do have a packed zip file, then you'll want to put it in the plugin folder so it gets synced.

Got it! I will set the default path to `.obsidian/icons/`. Thanks!

**Comment by lishid** [comment]
If you're interested, I suggest possibly a way to pack and unpack through zips using [this zip library](https://www.npmjs.com/package/@zip.js/zip.js). It will unfortunately bloat up your plugin's minified size by a few hundred kilobytes, but it may be worth considering.

See how we use it here:
https://github.com/obsidianmd/obsidian-importer/blob/master/src/zip.ts#L1

**Comment by FlorianWoelki** [comment]
> https://github.com/obsidianmd/obsidian-importer/blob/master/src/zip.ts#L1

Thank you for the suggestion. Zipping of the individual icon packs is already done, so that one icon pack is in one zip file. I think the safe approach would be to just set the default icon pack path to `.obsidian/path`

**Comment by FlorianWoelki** [comment]
> I apologize, but I am not following the solution. Do I need to zip up my icons and place them in `.obsidian/icons`? Or will it work without zipping them up? 

Please follow up with the question in the other issue. This issue wasn't intended to be about the syncing issue.

**Comment by FlorianWoelki** [comment]
> Thanks for the reply @FlorianWoelki . However:
> 
> 1. @lishid closed the other issue
> 2. And you specifically mention `syncing` in the original post of this issue
> 
> > Due to the new syncing mechanisms, all icon packs will have to be zipped into a file so that the `syncing` works smoothly.

Yes, but it still can be discussed there. Maybe you can also ask the question in #52.

**Comment by FlorianWoelki** [comment]
> Ok I will not discuss the syncing here. However, the issue of whether we need to zip up our custom icons would seem appropriate to discuss here.
> 
> I just deleted all custom icons from both computer and phone. I then zipped up my icons and put them in `.obsidian/plugins/iconize/icons`. I then told Iconize to look in in tis dir but it wouldn't recognize the pack. Am I doing something wrong?

You only need to zip your custom icon packs. That's how I would do it:
1. Add a custom icon pack and add all the icons to this pack
2. Zip your custom icon pack where it has been created
3. Enjoy the custom icon pack

**Comment by FlorianWoelki** [comment]
> My apologies again, but I don't understand. I think there might be a language barrier in our communication. I have tried the instructions above but it does not work. I don't understand why I would create a custom folder, then zip them up inside of that folder. I am confused
> 
> 
> 
> I have the following questions:
> 
> 1. Where am I supposed to tell Iconize to look for my icons? 
> 
> 2. When I zip up my icons, is the zip file supposed to be in the folder that previously contained them? For example
> 
> `.obsidian/plugins/iconize/icons/custom_icon_pack/zipped_icon_pack.zip`

No, you only need to zip the icon pack itself. At the moment, the plugin works through checking the icons in the zip file.

**Comment by FlorianWoelki** [comment]
> In my screenshot above, I made sure to show you that the icon pack is zipped, but the plugin won't recognize it.

I am going to take a closer look at this issue.

**Comment by YannMiro** [comment]
Resuming #272 here. Very interesting article and works fine on the plugin too 😃

https://nsisodiya.medium.com/hacking-font-awesome-for-custom-svg-icons-ea762118fa7b

**Comment by YannMiro** [comment]
A follow-up on #272. I don't know if the [Icomoon](https://icomoon.io/) solution has been mentioned before, but you can generate custom SVGs using the Font Awesome font library and they seem to be working seamlessly. The only issue I have in the past 9-10 days of usage is probably that the whole vault will take some seconds more to load, but it was a large one already, so...

**Comment by simonfossom** [comment]
@FlorianWoelki can you give us update on this? 

I sync icons folder normally, don't add icons in settings but just add zip and all works fine. 👍

**Comment by NomarCub** [comment]
In my opinion, not zipping the icons, and just using folders with plain SVG files would make a lot of this easier. No need for zipping manually, nor automatically.
I'll copy this from my closed issue, as it was closed, and kinda duplicates this one: (https://github.com/FlorianWoelki/obsidian-iconize/issues/453)


> I would like to have icons stored as individual plain text files, instead of opaque zip archive blobs, to version and sync them more easily.
> I version my vault with git, and this would make that easier too.

> (I already use a plugin (this one, but it's unmaintained: https://github.com/aidenlx/obsidian-icon-shortcodes), which stores each icon as a file.)

**Comment by NickAnderegg** [comment]
I second @NomarCub's point. The zip solution is great for bundling icon packs with the plugin, but there's much less advantage to bundling up a user's icon pack locally!

**Comment by Thomas-Schmall** [comment]
I would also prefer syncing of plain files. I made a folder with a hundred selected icons, so there's no big overhead.
An additional struggle is the naming though... because Iconize removes all white-spaces and special characters. So you end up with duplicate files in your vault and combined with syncing it's just all a big mess. 
I've never seen this weird like renaming thing in other plugins - it's totally unclear to me why that should be necessary.

Anyways, the only clean way to sync and avoid duplicates is:
- Remove all special characters and white-spaces from the names
- Use a zip file
- Store it in the vault directly (as in: not in the `.obsidian` folder)
- Keep a different folder somewhere for working on the icons

Works so far.

**Comment by MarcinWArdentCode** [comment]
Is it just me, or it seems like a lot of it is an overkill?

Isn't a much more common use case to just pick an image that I would like to use as an icon for a given file or folder (e.g. from my downloads directory), have it downsized accordingly and assigned?

Sometimes you just want to assign something relevant, and installing further 200 icons to get 1 thing might not be **the** answer you are looking for.

**Comment by simonfossom** [comment]
Yeap. this is what I'm doing. But then you might just to a situation where it's easier to 
`cat ./obsidian-icons/*.svg > icons.svg` just add a line to add it as --var-icon-name 

easier. lower on the system. no JS. 

You can hook on dataview field or any other thing that has datafield already. I use templater select dropdown for name.
